airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/state/StreamManager.kt

```diff
@@ -24,6 +25,8 @@ data class StreamProcessingFailed(val streamException: Exception) : StreamResult
 
 data object StreamProcessingSucceeded : StreamResult
 
+data class CheckpointId(val id: Long)
```

Comment:
very minor thing (`value class` has 0 runtime overhead, vs `data class` actually instantiating a new object)
```suggestion
@JvmInline
value class CheckpointId(val id: Long)
```